fluss-server/src/main/java/com/alibaba/fluss/server/kv/KvTablet.java

```diff
@@ -267,12 +271,27 @@ public LogAppendInfo putAsLeader(
                                 KvRecordReadContext.createReadContext(kvFormat, fieldTypes);
                         ValueDecoder valueDecoder =
                                 new ValueDecoder(readContext.getRowDecoder(schemaId));
-
+                        RowMergeEngine rowMergeEngine = getRowMergeEngine(schema);
+                        /**
+                         * In theory, once configured, it can be ignored directly. In the future, we
+                         * can provide configurations to choose between ignoring or throwing
+                         * exceptions directly.
+                         */
+                        if (partialUpdater != null && rowMergeEngine.ignorePartialUpdate()) {
+                            throw new UnsupportedOperationException(
+                                    String.format(
+                                            "The current merge engine %s does not support partial updates.",
+                                            mergeEngine.getType()));
+                        }
                         int appendedRecordCount = 0;
                         for (KvRecord kvRecord : kvRecords.records(readContext)) {
                             byte[] keyBytes = BytesUtils.toArray(kvRecord.getKey());
                             KvPreWriteBuffer.Key key = KvPreWriteBuffer.Key.of(keyBytes);
                             if (kvRecord.getRow() == null) {
+                                // currently, all supported merge engine will ignore delete row.
+                                if (rowMergeEngine.ignoreDelete()) {
```

Comment:
We still need the nullable `MergeEngineType` in the `KvTablet` to skip deleting if we introduce `RowMerger` without `ignoreDelete()`.